exp/query/api.go

```diff
@@ -29,21 +27,20 @@ type API struct {
 
 	// svc provides the underlying search API via v1.WeaviateServer
 	// TODO(kavi): Split `v1.WeaviateServer` into composable `v1.Searcher` and everything else.
-	svc protocol.WeaviateServer
+	svc      protocol.WeaviateServer
+	appState *state.State
 }
 
 func NewAPI(
-	traverser *traverser.Traverser,
+	appState *state.State,
```

Comment:
I would prefer to avoid coupling `query.API` with this global `appState`. Main rationales

1. Will make it unclear what are the explicit dependencies that `query.API` needs. For e.g: `appState` wraps lots of other things.
2. Easy to mock (so easy to test) if we know clear explicit dependencies, e.g: if it accepts `appState`, we need to mock the whole `state.State` to do any unit test case which can make the tests fragile and hard (because appState is global and subject to change)

Reply:
I took some liberty to push this change. Happy to discuss Nate :)

Resolving this conversation to unblock merging.